client/src/js/components/bhCashBoxFinder.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * The cash box finder component
+ *
+ * @module components/bhCashBoxFinder
+ *
+ * @description a component to deal with cash box, it let you choose a cash box
+ * 
+ **/
+ 
+ //A cash box finder component controller
+ 
+ function cashBoxFinderController (CashboxService, $translate){
+ 	
+ 	var ctrl = this;
+ 	ctrl.types = [
+ 					{label : $translate.instant('CASH_BOX_TYPE.PRIMARY'), is_auxillary : 0, is_bank :  0},
+ 					{label : $translate.instant('CASH_BOX_TYPE.AUXILLARY'), is_auxillary : 1, is_bank : 0},
+ 					{label : $translate.instant('CASH_BOX_TYPE.BANK'), is_bank : 1, is_auxillary : 0}
+ 				 ];
+ 	ctrl.selectedType = ctrl.types[0];
+
+ 	function init () {
+ 		ctrl.selectedCash = null;
+ 		ctrl.cashBoxValue = null;
+	 	ctrl.session = {
+	 		state : 'finding'
+	 	}
+	 	fetchcash('?full=1&is_auxillary=' + ctrl.selectedType.is_auxillary + '&is_bank=' + ctrl.selectedType.is_bank);
```

Comment:
Instead of manually writing query strings, you should use Angular's $http parameter serializer.  [Here is the documentation](https://docs.angularjs.org/api/ng/service/$http).